client/js/options.js

```diff
@@ -91,6 +93,13 @@ settings.on("change", "input, select, textarea", function() {
 			// otherwise, users get notifications for everything
 			return h !== "";
 		});
+		// Construct regex with wordboundary for every highlight item
+		var highlightsTokens = options.highlights.map(function(h) {
```

Comment:
There's already a `map` with `trim` in it, you could probably throw `escapeRegExp` into it instead of iterating twice.